### ISSUE TITLE
Clearer Pytorch Examples

### ISSUE THREAD

**Issue by bhairavmehta95** [awslabs/amazon-sagemaker-examples#468]
It would be nice if the Pytorch examples followed some sort of structure, or were organized better. Normally, you have all the functions, and then the normal code that runs when you call the python file gets encapsulated within `if __name__ == '__main__':` (but really, there probably shouldn't be any free standing python code..)

 It's quite hard to follow, since functions are interlayed with normal python code, and they sort of interweave each other. This makes it extremely hard to see what gets executed where, and I found myself constantly checking the tabs / spacing to see if certain code was inside of a function, or just being executed.

Example file:
https://github.com/awslabs/amazon-sagemaker-examples/blob/master/sagemaker-python-sdk/pytorch_lstm_word_language_model/source/train.py

**Comment by laurenyu** [comment]
Thanks for the feedback! Totally understand where you're coming from - we're always reprioritizing our work based on customer feedback, and we'll look into improving the readability of the code when we can. We're also always open to contributions, so if you have any interest in editing the code to make it more readable for yourself and then submitting that in a PR, feel free to do so!

**Comment by nadiaya** [comment]
This example is written in that way because it was based on the facebook's example:
https://github.com/pytorch/examples/blob/master/word_language_model/main.py

One purpose of the example was to show how little modification is required to an existing pytorch code to make it work in SageMaker.
